contracts/SpeedMarkets/SpeedMarketsAMM.sol

```diff
@@ -71,39 +72,26 @@ contract SpeedMarketsAMM is Initializable, ProxyOwned, ProxyPausable, ProxyReent
     mapping(address => AddressSetLib.AddressSet) internal _activeMarketsPerUser;
     mapping(address => AddressSetLib.AddressSet) internal _maturedMarketsPerUser;
 
-    struct MarketData {
-        address user;
-        bytes32 asset;
-        uint64 strikeTime;
-        int64 strikePrice;
-        SpeedMarket.Direction direction;
-        uint buyinAmount;
-        bool resolved;
-        int64 finalPrice;
-        SpeedMarket.Direction result;
-        bool isUserWinner;
-        uint256 createdAt;
-    }
-
     mapping(address => bool) public whitelistedAddresses;
     IMultiCollateralOnOffRamp public multiCollateralOnOffRamp;
     bool public multicollateralEnabled;
 
     mapping(bytes32 => mapping(SpeedMarket.Direction => uint)) public maxRiskPerAssetAndDirection;
     mapping(bytes32 => mapping(SpeedMarket.Direction => uint)) public currentRiskPerAssetAndDirection;
 
-    struct Risk {
-        SpeedMarket.Direction direction;
-        uint current;
-        uint max;
-    }
-
     uint64 public maximumPriceDelayForResolving;
 
-    mapping(address => bool) private marketHasCreatedAtAttribute;
+    mapping(address => bool) public marketHasCreatedAtAttribute;
 
     address public referrals;
 
+    uint[] public timeThresholdsForFees;
+    uint[] public lpFees;
+
+    SpeedMarketsAMMUtils public speedMarketsAMMUtils;
+
+    mapping(address => bool) public marketHasFeeAttribute;
```

Comment:
does it really need to be public? a public method adds an automatic getter which increases contract size

Reply:
marketHasFeeAttribute is read from the contract SpeedMarketsAMMData, but speedMarketsAMMUtils should be private